pkg/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1473,6 +1473,64 @@ func TestTombstoneUpdate(t *testing.T) {
 	dst.CheckQueryResults(t, "SELECT * FROM tab WHERE pk = 1", [][]string{})
 }
 
+func TestReplicateComputedColumns(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	skip.UnderDeadlock(t)
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	server, s, dbSource, dbDest := setupLogicalTestServer(t, ctx, testClusterBaseClusterArgs, 1)
+	defer server.Stopper().Stop(ctx)
+
+	// Note: the primary key is basicaly (a, b), but with a computed column that
+	// is d=a+b.
+	createTableStmt := `CREATE TABLE computed_cols (
+		a INT,
+		b INT,
+		c INT,
+		d INT AS (a + b) STORED,
+		cIsOdd BOOLEAN AS (c % 2 = 1) VIRTUAL,
+		PRIMARY KEY (a, b)
```

Comment:
could also be worth testing a table with a computed column in the pk

Reply:
oh, i may need to allow virtual cols in the pk first. i can modify this test in a later patch.

Reply:
I meant to use column `d` for this. The PK is (a, b) from the time I spent trying to get to the bottom of the optimistic insert corruption.